### PR TITLE
[ENG-1252] - fixed add wallet permission & ts module declaration issues

### PR DESCRIPTION
**Pull request by dhruv-kumar-jha**

### CODE REVIEW

File: src/user/user.ts [hunk 1]
```diff
@@ -120,7 +120,7 @@ export class User extends Observable<SubscriberUser> {
       throw new Error('AccessToken is missing or has expired')
     }
     // get account specified in access token
-    const account = this?._accountName
+    const account = this?._accountName || this?._accessTokenHelper.accountName
```
Comment (dhruv-kumar-jha) [inline src/user/user.ts:123]: @traylewin Sometimes `this._accountName` is null but it's present in `this?._accessTokenHelper.accountName`
This change fixes that issue.

File: src/user/user.ts [hunk 2]
```diff
@@ -239,14 +239,14 @@ export class User extends Observable<SubscriberUser> {
     permissions: WalletPermission[]
     walletType: ExternalWalletType
   }) {
+    // get latest user info
+    await this.getData()
```
Comment (dhruv-kumar-jha) [inline src/user/user.ts:243]: This change fixes the flow where user (in oreid-portal) tried to connect their wallet and it threw error `Error: Public Key of signed string not found in wallet` - Tested for `ethereum` chain

**Flow**
- In oreid-portal, Select a dApp from "My Dapps"
- Click on "Connect Wallet"
- Login to the dApp
- Connect the wallet & sign string
- FAILS THE FIRST TIME
- try again - this time it works

The root issue was it failed at condition `if (!this.accountName || isNullOrEmpty(permissions)` as `this.accountName` was undefined/null - calling `await this.getData()` prior to this condition fixes the issue

**Review by yardz — APPROVED** [review]
LGTM

File: src/user/user.ts (hunk 1 above)
Comment (traylewin) [reply src/user/user.ts:123]: This doesn't solve the root of the problem it just hacks around it and adds more future confusion.
I would guess the solution is to add code in onUpdateAccessTokenHelper() to update this.accountName from accessTokenHelper.accountName. revert this code and add the fix there like...
  private onUpdateAccessTokenHelper = (newAccessTokenHelper: AccessTokenHelper) => {
    this._accountName = newAccessTokenHelper.accountName
    super.callSubscribers()
  }

**Review by traylewin — CHANGES_REQUESTED** [review]
Please fix as requested

**Pull request merged**